android/src/main/java/com/reactnativereceivesharingintent/RNMediaInfoUtils.java

```diff
@@ -0,0 +1,197 @@
+package com.reactnativereceivesharingintent;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.database.Cursor;
+import androidx.exifinterface.media.ExifInterface;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.provider.OpenableColumns;
+
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.WritableNativeMap;
+
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RNMediaInfoUtils {
+
+    private static final String TAG = "MEDIA_INFO_UTILS";
+
+    private static int getExifOrientation(String path,Boolean isImage) {
+        try {
+            if(!isImage){
+                Log.d(TAG, "Media is not an image. Returning ORIENTATION_UNDEFINED.");
+                //If media is not image, return fallback value
+                return ExifInterface.ORIENTATION_UNDEFINED;
+            }
+            ExifInterface exif = new ExifInterface(path);
+            String orientation = exif.getAttribute(ExifInterface.TAG_ORIENTATION);
+            if(orientation == null){
+                Log.d(TAG, "Orientation is null. Returning ORIENTATION_UNDEFINED.");
+                //If media is not image, return fallback value
+                return ExifInterface.ORIENTATION_UNDEFINED;
+            }
+            Log.d(TAG, "Exif orientation: " + orientation);
+            return Integer.parseInt(orientation);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Log.e(TAG, "Error getting exif orientation", e);
+            return ExifInterface.ORIENTATION_UNDEFINED;
+        }
+    }
+
+    private static Map<String, String> getMediaInfo(Context context, Uri contentUri) {
+        Map<String, String> resultMap = new HashMap<>();
+        ContentResolver contentResolver = context.getContentResolver();
+        try {
+            Cursor queryCursor = contentResolver.query(contentUri, null, null, null, null);
+
+            if (queryCursor != null) {
+                queryCursor.moveToFirst();
+                int displayNameIndex = queryCursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
+                int sizeIndex = queryCursor.getColumnIndex(OpenableColumns.SIZE);
+                int dateTakenIndex = queryCursor.getColumnIndex(MediaStore.MediaColumns.DATE_TAKEN);
+                int dateModifiedIndex = queryCursor.getColumnIndex(MediaStore.MediaColumns.DATE_MODIFIED);
+
+                if (displayNameIndex != -1) {
```

Comment:
I'm curious are all these checks needed? What happens if you try to `.put` when this condition is false?

Reply:
From the doc:

The result and whether this method throws an exception when the column value is null or the column type is not a string type is implementation-defined. 

So what I understood: "It can throw an error or not"